### PR TITLE
remove the last use of PrincipalIdealDomain

### PR DESCRIPTION
**Pull request by fchapoton**
This is removing the last use of the auld class `PrincipalIdealDomain`

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

**Comment by vincentmacri** [comment]
If this class isn't used anymore, can the class itself be deleted entirely? It looks like it's already deprecated.

**Comment by fchapoton** [comment]
I has only be deprecated formally recently. We have to wait one year after the first release after the deprecation.

File: src/sage/rings/polynomial/polynomial_ring.py [hunk 1]
```diff
@@ -2217,9 +2216,14 @@ def _element_class():
             from sage.rings.polynomial.polynomial_element_generic import Polynomial_generic_dense_field
             return Polynomial_generic_dense_field
 
+        if category is None:
+            cat = PrincipalIdealDomains()
+        else:
+            cat &= PrincipalIdealDomains()
```
Comment (mantepse) [inline src/sage/rings/polynomial/polynomial_ring.py]: There is not `cat` here yet, is there?

Comment (fchapoton) [reply src/sage/rings/polynomial/polynomial_ring.py]: oups, désolé

**Review by mantepse — APPROVED** [review]
Looks correct!

**Pull request merged**